lib/notifier/index.js

```diff
@@ -0,0 +1,424 @@
+  }
+
+  const age = Math.floor(lastEvent.age)
+  if (age > STALE_STATE) {
+    return [{ code: STALE, state, age, machineName }]
+  }
+
+  return []
+}
+
+async function transactionNotify (tx, rec) {
+  const settings = await settingsLoader.loadLatest()
+  const notifSettings = configManager.getGlobalNotifications(settings.config)
+  const highValueTx = tx.fiat.gt(notifSettings.highValueTransaction || Infinity)
+  const isCashOut = tx.direction === 'cashOut'
+  
+  // high value tx on database
+  if(highValueTx && tx.direction === 'cashIn' || highValueTx && tx.direction === 'cashOut' && rec.isRedemption) {
+    const direction = tx.direction === "cashOut" ? 'cash-out' : 'cash-in'
+    const message = `${tx.fiat} ${tx.fiatCode} ${direction} transaction`
+    const detailB = utils.buildDetail({deviceId: tx.deviceId, direction, fiat: tx.fiat, fiatCode: tx.fiatCode, cryptoAddress: tx.toAddress})
+    queries.addNotification(HIGH_VALUE_TX, message, detailB)
+  }
+  
+  // alert through sms or email any transaction or high value transaction, if SMS || email alerts are enabled
+  const cashOutConfig = configManager.getCashOut(tx.deviceId, settings.config)
+  const zeroConfLimit = cashOutConfig.zeroConfLimit
+  const zeroConf = isCashOut && tx.fiat.lte(zeroConfLimit)
+  const notificationsEnabled = notifSettings.sms.transactions || notifSettings.email.transactions
+  const customerPromise = tx.customerId ? customers.getById(tx.customerId) : Promise.resolve({})
+  
+  if (!notificationsEnabled && !highValueTx) return Promise.resolve()
+  if (zeroConf && isCashOut && !rec.isRedemption && !rec.error) return Promise.resolve()
+  if (!zeroConf && rec.isRedemption) return sendRedemptionMessage(tx.id, rec.error)
+
+  return Promise.all([
+    machineLoader.getMachineName(tx.deviceId),
+    customerPromise
+  ])
+  .then(([machineName, customer]) => {
+    return utils.buildTransactionMessage(tx, rec, highValueTx, machineName, customer)
+  })
+  .then(([msg, highValueTx]) => sendTransactionMessage(msg, highValueTx))
+}
+
+function sendRedemptionMessage(txId, error) {
+  const subject = `Here's an update on transaction ${txId}`
+  const body = error
+    ? `Error: ${error}`
+    : 'It was just dispensed successfully'
+
+  const rec = {
+    sms: {
+      body: `${subject} - ${body}`
+    },
+    email: {
+      subject,
+      body
+    }
+  }
+  return sendTransactionMessage(rec)
+}
+
+async function sendTransactionMessage(rec, isHighValueTx) {
+  const settings = await settingsLoader.loadLatest()
+  const notifications = configManager.getGlobalNotifications(settings.config)
+
+  let promises = []
+
+  const emailActive =
+    notifications.email.active &&
+    (notifications.email.transactions || isHighValueTx)
+  if (emailActive) promises.push(emailFuncs.sendMessage(settings, rec))
+
+  const smsActive =
+    notifications.sms.active &&
+    (notifications.sms.transactions || isHighValueTx)
+  if (smsActive) promises.push(smsFuncs.sendMessage(settings, rec))
```

Comment:
If you are using `.push()` on an array, you can initialize it with `const`. That prevents reassignment of the array, but allows for push and pop operations to occur. See snippet on image 
![image](https://user-images.githubusercontent.com/18738838/104453717-bc755f80-559c-11eb-8269-725ab752262c.png)


```suggestion
  const promises = []

  const emailActive =
    notifications.email.active &&
    (notifications.email.transactions || isHighValueTx)
  if (emailActive) promises.push(emailFuncs.sendMessage(settings, rec))

  const smsActive =
    notifications.sms.active &&
    (notifications.sms.transactions || isHighValueTx)
  if (smsActive) promises.push(smsFuncs.sendMessage(settings, rec))
```